### DISCUSSION TITLE
Creating time series with Python SDK

### DISCUSSION THREAD

**Discussion by asmunds** [Volue-Public/energy-mesh-python#382] [Ideas]
Hi,
We would really benefit if it was possible to create time series with Python SDK.
Is there any plan to add this functionality?

**Comment by tnoczyns-volue** [comment]
Hi,

By creating time series you mean:
* creating physical time series, where you can write new data?
* connecting existing physical time series to time series attributes?
* changing calculation expressions in time series calculations?
* something else?

See: https://volue-public.github.io/energy-mesh-python/timeseries.html

Thanks,
Tomasz

**Reply by asmunds** [reply]
I mean "creating physical time series, where you can write new data"

**Reply by torsannes** [reply]
Hi,

Just to let me understand this correctly, these time series that you want to create are not related to any object in the Mesh model, but only exists as "free" time series in the Oracle database?

BR, Tor

**Reply by asmunds** [reply]
No, I want to create the time series (and I guess the required attribute for it) on an object in the Mesh model.

**Comment by torsannes** [comment]
Good input, @asmunds!
I will put this on the activity list to be prioritized for the next release (2024.2) and it includes also an increase in the gRPC interface for Mesh as well.
The following issues are created:
* Mesh: https://github.com/Volue/energy-mesh/issues/4605
* Python: https://github.com/Volue-Public/energy-mesh-python/issues/383

**Reply by asmunds** [reply]
Så bra!

**Comment by asmunds** [comment]
Is there a way to connect a physical timeseries to a time series attribute with the API? I couldn't figure out how by reading the documentation.

**Reply by tnoczyns-volue** [reply]
Yes, there is. Please check `working_with_model` example:
https://github.com/Volue-Public/energy-mesh-python/blob/d11d51198e5e9fc09129a40f3eb9063595a911f9/src/volue/mesh/examples/working_with_model.py#L68-L72

API docs:
https://volue-public.github.io/energy-mesh-python/api.html#volue.mesh.Connection.Session.update_timeseries_attribute

BTW physical time series is a time series resource. See more info:
https://volue-public.github.io/energy-mesh-python/timeseries.html#types

**Reply by asmunds** [reply]
Okay, thanks. But what if you need to connect an already existing physical time series, and you only have its path and name, but not its timeseries_resource_key? Is there a function for searching for an unconnected time series resource with the API?

**Reply by asmunds** [reply]
I can see that its almost possible, because when i do this:

`session.get_object(target="Resource"+path+name)`

I get:

```
grpc._channel._InactiveRpcError: <_InactiveRpcError of RPC that terminated with:
	status = StatusCode.INVALID_ARGUMENT
	details = "invalid type of object Timeseries("Resource/prodrisk/test__test_module_ReservoirVolume_mean")"
	debug_error_string = "UNKNOWN:Error received from peer  {created_time:"2025-06-02T12:54:35.7830987+00:00", grpc_status:3, grpc_message:"invalid type of object Timeseries(\"Resource/prodrisk/test__test_module_ReservoirVolume_mean\")"}"
>
```

so it recognizes the item at the path as a Timeseries

**Reply by asmunds** [reply]
I guess I am missing a get_timeseries-method

**Reply by tnoczyns-volue** [reply]
Unfortunately we don't have API in Python SDK for listing/getting time series resources based on path or name.
You can use other applications to get time series key(s), e.g. Time Series application. Is this an option?
